src/CodeGen_C.cpp

```diff
@@ -526,14 +555,22 @@ class CppVector {
         }
     }
 
-    static Vec shuffle(const Vec &a, const int32_t indices[Lanes]) {
+    template<int N, int First, int Second, int... Rest>
+    static void shuffle_impl(Vec &dst, const Vec &src) {
```

Comment:
It might be helpful to have a few inlined versions of small numbers of indices, to reduce the recursion depth. I found this improved intermediate code size/compile time in similar code.